tap_campaign_monitor/client.py

```diff
@@ -37,9 +47,17 @@ def get_timezone(self):
 
         return tap_campaign_monitor.timezones.from_string(timezone)
 
-    def make_request(self, url, method, base_backoff=30,
-                     params=None, body=None):
-
+    @backoff.on_exception(
+        backoff.expo,
+        (ConnectionError, Server5xxError, Server429Error),
```

Comment:
429 is rate limiting, should be handled in separate backoff where you might want to wait longer. See if you can find appropriate value from the documentation.